src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -0,0 +1,45 @@
+#if NETCOREAPP
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    /// <summary>
+    /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+    /// </summary>
+    public partial class Startup
+    {
+        internal static System.Runtime.Loader.AssemblyLoadContext DependencyLoadContext { get; } = new ManagedProfilerAssemblyLoadContext();
+
+        private static string ResolveManagedProfilerDirectory()
+        {
+            string tracerFrameworkDirectory = "netstandard2.0";
+            var tracerHomeDirectory = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME") ?? string.Empty;
+            return Path.Combine(tracerHomeDirectory, tracerFrameworkDirectory);
+        }
+
+        private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sender, ResolveEventArgs args)
+        {
+            string assemblyName = new AssemblyName(args.Name).Name;
+            var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName}.dll");
+
+            if (assemblyName.ToLower().Contains("datadog.trace") && File.Exists(path))
```

Comment:
Would it be better to use our public key(s)?

Reply:
Yeah that's a good idea. I expect we'll always be loading with the full name, like below, so `args.Name` should contain the full name

Reply:
Added in 1e892a61569c1df2ec49af867ae9920e2c70aba7